### ISSUE TITLE
Customizing CMake plugin project settings

### ISSUE THREAD

**Issue by zenera** [eranif/codelite#1318]
I'm a newbie for codelite and cmake. But I feel codelite has some bugs for cmakeplugin settings. I'm currently using codelite 9.1.8 on Ubuntu 16.04(64bit). I have read about CMakePlugin document for CodeLite 9.1.7 in http://codelite.org/LiteEditor/TheCMakePlugin.

Currently, my working settings for cmake plugin is to use cmake as a custom target.
It works fine for customized working directory for executing cmake, building and running a project.
Although it is still confused in setting cmake plugin to work correctly.

For example, the working directory setting is duplicated in Project Settings (Customize > Working Directory and General > Intermediate Folder). But, if I want to run a built executable I have to set General > Executable to Run/Debug.

Newly introduced "Run CMake" project context menu will be great if I could customize working directory. Then the customized taget for cmake would be useless. All the settings for cmake pluin could be done in Project Settings > General tab.

Usually I make each project as a sub folder in a workspace. But it seemed "Run CMake" menu assumes a workspace to be a project. Furthermore it doesn't allow working in customized working directory, right?
And a CMakeLists.txt for a workspace is mandatory. All the restrictions here are intended features and not bugs?

In summary, I'd like to welcome the "Run CMake" project context menu if it supports customized project settings.

Furthemore, relevant menu items for cmakeplugin needs to be grouped:  Export CMakeLists.txt, Run CMake, Open CMakeLists.txt.

One more bug, $(WorkspaceConfiguration) macro doesn't work when used in Projet Settings > Customize > Working Directory. $(ConfigurationName) macro works fine though.

Thanks for great tools.

**Comment by eranif** [comment]
Thanks for the feedback on the CMake plugin changes.
HOWEVER, when using "Custom Project" the CMake plugin does not work. period.
When using a custom project it tells CodeLite that **YOU** take the ownership of building the project (this means: generate proper CMake, QMake etc file and convert them Makefile (or ...))

This will not going to change.
If you have a concrete comments on how the current plugin work (without "custom build" enabled) - please provide them

**Comment by zenera** [comment]
My comments were assumed with enabling custom build.

I can't understand what you said. Are there any methods importing external cmake-based projects without enabling custom build in CodeLite?

What I meant was about merging "Custom Targets... > cmake" and "Run CMake" in project context menu. Specifically working directory could be set something like $(ProjectPath)/build-$(ConfigurationName).
